File: 1010_Pairs of Songs With Total Durations Divisible by 60/PairsofSongsWithTotalDurationsDivisibleby60.cs
public class Solution {
    public int NumPairsDivisibleBy60(int[] time) {
        var ans = 0;
        var dict = new Dictionary<int,int>();
        
        foreach(var A in time)
        {
            var B = (60 - A % 60) % 60;
            if(dict.ContainsKey(B))
                ans += dict[B];
            if(!dict.ContainsKey(A%60)) dict.Add(A%60,0);
            dict[A%60]+=1;
        }
        
        return ans;
    }
}

File: 1010_Pairs of Songs With Total Durations Divisible by 60/PairsofSongsWithTotalDurationsDivisibleby60.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        if(time == null || time.length <= 1) return 0;
        int ans = 0;
       
        int[] mods = new int[60];
        
        for(int v : time) {
            int mod = (60 - v % 60) % 60;
            ans += mods[mod];
            mods[v % 60]++;
        }
        
        return ans;
    }
}